src/mono/System.Private.CoreLib/src/System/Enum.Mono.cs

```diff
@@ -9,24 +9,44 @@ namespace System
     public partial class Enum
     {
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern bool GetEnumValuesAndNames(RuntimeType enumType, out ulong[] values, out string[] names);
+        private static extern bool GetEnumValuesAndNames(QCallTypeHandle enumType, out ulong[] values, out string[] names);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern object InternalBoxEnum(RuntimeType enumType, long value);
+        private static extern void InternalBoxEnum(QCallTypeHandle enumType, ObjectHandleOnStack res, long value);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private extern CorElementType InternalGetCorElementType();
+        private static extern CorElementType InternalGetCorElementType(QCallTypeHandle enumType);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern RuntimeType InternalGetUnderlyingType(RuntimeType enumType);
+        private static extern void InternalGetUnderlyingType(QCallTypeHandle enumType, ObjectHandleOnStack res);
+
+        private static object InternalBoxEnum(RuntimeType enumType, long value)
+        {
+            object? res = null;
```

Comment:
Wondering if it'd be better to use
```c#
object res = null!;
```

for cases where you expect res to never be null and user `object? res = null` where it can be.

Reply:
res is null initially, so object! looks like a better type for it.